Further porting of high frequency IPC types to the new serialization format

gavin-apple opened the pull request:
#### 83f96fe0a90a9d84cf00e00e38bc6b713666dbd6
<pre>
Further porting of high frequency IPC types to the new serialization format
<a href="https://bugs.webkit.org/show_bug.cgi?id=249147">https://bugs.webkit.org/show_bug.cgi?id=249147</a>
rdar://103253106

Reviewed by Alex Christensen.

This change ports more types to the new serialization format. It
includes:
    - MediaRecorderPrivateOptions
    - FloatQuad
    - ApplePayErrorContactField
    - ApplePayErrorCode
    - SelectionRenderingBehavior
    - UserVerificationRequirement
    - CDMKeySystemConfiguration
    - MediaConfiguration
    - MediaEncodingConfiguration
    - MediaDecodingConfiguration

* Source/WebCore/Modules/applepay/ApplePayErrorCode.h:
* Source/WebCore/Modules/applepay/ApplePayErrorContactField.h:
* Source/WebCore/Modules/webauthn/UserVerificationRequirement.h:
* Source/WebCore/platform/encryptedmedia/CDMKeySystemConfiguration.h:
(WebCore::CDMKeySystemConfiguration::encode const): Deleted.
(WebCore::CDMKeySystemConfiguration::decode): Deleted.
* Source/WebCore/platform/graphics/FloatQuad.h:
(WebCore::FloatQuad::encode const): Deleted.
(WebCore::FloatQuad::decode): Deleted.
* Source/WebCore/platform/mediacapabilities/MediaConfiguration.h:
(WebCore::MediaConfiguration::encode const): Deleted.
(WebCore::MediaConfiguration::decode): Deleted.
* Source/WebCore/platform/mediacapabilities/MediaDecodingConfiguration.h:
(WebCore::MediaDecodingConfiguration::encode const): Deleted.
(WebCore::MediaDecodingConfiguration::decode): Deleted.
* Source/WebCore/platform/mediacapabilities/MediaEncodingConfiguration.h:
(WebCore::MediaEncodingConfiguration::encode const): Deleted.
(WebCore::MediaEncodingConfiguration::decode): Deleted.
* Source/WebCore/platform/mediarecorder/MediaRecorderPrivateOptions.h:
(WebCore::MediaRecorderPrivateOptions::encode const): Deleted.
(WebCore::MediaRecorderPrivateOptions::decode): Deleted.
* Source/WebKit/Scripts/webkit/messages.py:
(serialized_identifiers):
* Source/WebKit/Shared/Cocoa/WebCoreArgumentCodersCocoa.serialization.in:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/WebAuthentication/Cocoa/LocalConnection.h:

Canonical link: <a href="https://commits.webkit.org/257847@main">https://commits.webkit.org/257847@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c04399c99a8618df4a75f3614e145d72ecb5dff4

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/100206 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/9373 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/33282 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/109533 "Built successfully") | [❌ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/169766 "Failed to checkout and rebase branch from PR 7489") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/104198 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/10274 "Built successfully") | [⏳ 🛠 mac-AS-debug ](https://ews-build.webkit.org/#/builders/macOS-AppleSilicon-Ventura-Debug-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/92627 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/107422 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/105982 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/7774 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/91045 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/34455 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/103719 "Passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/89691 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/22440 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/77399 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/3132 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/23956 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/3110 "Built successfully") | [⏳ 🧪 mac-AS-debug-wk2 ](https://ews-build.webkit.org/#/builders/macOS-AppleSilicon-Ventura-Debug-Build-EWS "Waiting in queue, processing has not started yet") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/9238 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/43444 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5388 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/4942 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
```diff
@@ -1918,3 +1952,42 @@ struct WebCore::GraphicsContextGLAttributes {
+};
+#endif // ENABLE(WEB_AUTHN)
+
+struct WebCore::CDMKeySystemConfiguration {
```
This needs #if ENABLE(ENCRYPTED_MEDIA)

gavin-apple committed:
Further porting of high frequency IPC types to the new serialization format
https://bugs.webkit.org/show_bug.cgi?id=249147
rdar://103253106

Reviewed by Alex Christensen.

This change ports more types to the new serialization format. It
includes:
    - MediaRecorderPrivateOptions
    - FloatQuad
    - ApplePayErrorContactField
    - ApplePayErrorCode
    - SelectionRenderingBehavior
    - UserVerificationRequirement
    - CDMKeySystemConfiguration
    - MediaConfiguration
    - MediaEncodingConfiguration
    - MediaDecodingConfiguration

* Source/WebCore/Modules/applepay/ApplePayErrorCode.h:
* Source/WebCore/Modules/applepay/ApplePayErrorContactField.h:
* Source/WebCore/Modules/webauthn/UserVerificationRequirement.h:
* Source/WebCore/platform/encryptedmedia/CDMKeySystemConfiguration.h:
(WebCore::CDMKeySystemConfiguration::encode const): Deleted.
(WebCore::CDMKeySystemConfiguration::decode): Deleted.
* Source/WebCore/platform/graphics/FloatQuad.h:
(WebCore::FloatQuad::encode const): Deleted.
(WebCore::FloatQuad::decode): Deleted.
* Source/WebCore/platform/mediacapabilities/MediaConfiguration.h:
(WebCore::MediaConfiguration::encode const): Deleted.
(WebCore::MediaConfiguration::decode): Deleted.
* Source/WebCore/platform/mediacapabilities/MediaDecodingConfiguration.h:
(WebCore::MediaDecodingConfiguration::encode const): Deleted.
(WebCore::MediaDecodingConfiguration::decode): Deleted.
* Source/WebCore/platform/mediacapabilities/MediaEncodingConfiguration.h:
(WebCore::MediaEncodingConfiguration::encode const): Deleted.
(WebCore::MediaEncodingConfiguration::decode): Deleted.
* Source/WebCore/platform/mediarecorder/MediaRecorderPrivateOptions.h:
(WebCore::MediaRecorderPrivateOptions::encode const): Deleted.
(WebCore::MediaRecorderPrivateOptions::decode): Deleted.
* Source/WebKit/Scripts/webkit/messages.py:
(serialized_identifiers):
* Source/WebKit/Shared/Cocoa/WebCoreArgumentCodersCocoa.serialization.in:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/WebAuthentication/Cocoa/LocalConnection.h:

Canonical link: https://commits.webkit.org/257847@main
